spec-main/api-app-path-spec.ts

```diff
@@ -0,0 +1,315 @@
+    it('by default', async () => {
+      const output = await runTestApp('app-custom-path');
+      switch (process.platform) {
+        case 'darwin':
+          expect(output.appLogs).to.equal(path.join(os.homedir(), 'Library', 'Logs', 'Electron'));
+          break;
+        case 'win32':
+        default:
+          expect(output.appLogs).to.equal(path.join(output.appData, defaultAppName, 'logs'));
+          expect(output.appLogs).to.equal(path.join(output.userData, 'logs'));
+          break;
+      }
+    });
+
+    // Linux
+    // app path module setAppLogsPath() setAppLogsPath(/tmp/mylogs) - setAppLogsPath(/tmp/mylogs)
+    // /home/builduser/project/src/electron/spec-main/api-app-path-spec.ts
+    // AssertionError: expected '/home/builduser/.config/app-custom-path/logs' to equal '/tmp/mylogs'
+
+    // if (process.platform === 'win32') {
+    //   it(`setAppLogsPath('${appLogsPath}')`, async () => {
+    //     const output = await runTestApp('app-custom-path', `-custom-applogs=${appLogsPath}`)
+    //     expect(output.appLogs).to.equal(appLogsPath)
+    //   })
+    // }
+
+    // To review later
+    it('setAppLogsPath()', async () => {
+      const defaultAppLogs = app.getPath('logs');
+      app.setAppLogsPath(appLogsPath);
+      expect(app.getPath('logs')).to.equal(appLogsPath);
+      app.setAppLogsPath();
+      expect(app.getPath('logs')).to.equal(defaultAppLogs);
+    });
+
+    it('throws when a relative path is passed', () => {
+      const badPath = 'hey/hi/hello';
+
+      expect(() => {
+        app.setAppLogsPath(badPath);
+      }).to.throw(/Path must be absolute/);
+    });
+  });
+
+  //  describe('getPath("logs")', () => {
+  //    const logsPaths = {
+  //      'darwin': path.resolve(os.homedir(), 'Library', 'Logs'),
+  //      'linux': path.resolve(app.getPath('userData'), 'logs'),
+  //      'win32': path.resolve(app.getPath('userData'), 'logs'),
+  //    }
+
+  //    it('has no logs directory by default', () => {
+  //      const osLogPath = (logsPaths as any)[process.platform]
+  //      expect(fs.existsSync(osLogPath)).to.be.false()
+  //    })
+
+  //    it('creates a new logs directory if one does not exist', () => {
+  //      expect(() => { app.getPath('logs') }).to.not.throw()
+
+  //      const osLogPath = (logsPaths as any)[process.platform]
+  //      expect(fs.existsSync(osLogPath)).to.be.true()
+  //    })
+  //  })
+});
+
+async function runTestApp (name: string, ...args: any[]) {
+  const appPath = path.join(fixturesPath, name);
+  const electronPath = process.execPath;
+  const appProcess = cp.spawn(electronPath, [appPath, ...args]);
+
+  let output = '';
+  appProcess.stdout.on('data', (data) => { output += data; });
+
+  await emittedOnce(appProcess.stdout, 'end');
+
+  // console.log(`ouput=${output}`)
+  try {
+    const outputJSON = JSON.parse(output);
+    return outputJSON as Payload;
+  } catch (err) {
+    console.log(`Error ${err}) with ${output}`);
+    throw err;
+  }
+}
```

Comment:
This util is very similar to one in spec-main/api-app-spec.ts.

Maybe it should be extracted into spec-helpers.ts where the same code can be used by both?

Reply:
In fact, there is another duplicate in api-ipc-main-spec.ts#51. If you do not mind, would prefer to focus on the main feature.